### PR TITLE
Double multiselect filter

### PR DESCRIPTION
**Pull request by JonTheBurley**
This PR adds a Double Multi Select Filter module. It also enhances the Filter to allow for overriding the FilterSectionHeader render method.

image

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -46,8 +46,9 @@ We've added storybook pages to show how each component can be used.
 * [Button](https://toyotaresearchinstitute.github.io/lakefront/?path=/docs/lakefront-button--all-buttons)
 * [Checkbox](https://toyotaresearchinstitute.github.io/lakefront/?path=/docs/lakefront-checkbox--checkbox)
 * [Collapsible](https://toyotaresearchinstitute.github.io/lakefront/?path=/docs/lakefront-collapsible--collapsible)
-* [Filter](https://toyotaresearchinstitute.github.io/lakefront/?path=/docs/lakefront-filter--all-filters)
+* [Filter](https://toyotaresearchinstitute.github.io/lakefront/?path=/docs/lakefront-filter-allfilters--no-filter-bar)
```
Comment (jason-ransom-slalom) [inline README.md:49]: Thanks for fixing!

File: src/Filter/modules/DoubleMultiSelectFilter/DoubleMultiSelectFilter.tsx [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+    DoubleMultiSelectFilterOptions,
+    FilterModule
+} from 'src/Filter/types';
+import DoubleMultiSelect from './DoubleMultiSelect';
+import { FilterSectionHeader } from 'src/Filter/components';
+import FilterValueChips from 'src/Filter/components/FilterSectionHeader/FilterValueChips';
+import { FilterBadge } from 'src/Filter/components/FilterSectionHeader/filterSectionHeaderStyles';
+import styled from '@emotion/styled';
+
+const FilterCount = styled.div(({ theme }) => ({
```
Comment (jason-ransom-slalom) [inline src/Filter/modules/DoubleMultiSelectFilter/DoubleMultiSelectFilter.tsx:13]: Can you add these styles to it's own style utils to be inline to other filters (thank @zach-jones-slalom for this one) 😂

File: src/Filter/types/index.ts [hunk 3]
```diff
@@ -80,7 +80,12 @@ export interface FilterModule<T> {
     /**
      * Renders the filter input controls in the left filter drawer.
      */
-    renderComponent(input: FilterRenderProps<T>): React.ReactElement;
+    renderComponent(input: FilterRenderProps<T>): ReactElement;
+    /**
+     * Overrides the default FilterSectionHeader for a filter module. Recommended usage is the provide a customized
```
Comment (jason-ransom-slalom) [inline src/Filter/types/index.ts:85]: Can you update to "Recommended usage is ***to*** provide a customized..."

**Review by jason-ransom-slalom — CHANGES_REQUESTED** [review]
Looks good! Please add to the filterPageUtil.tsx so that it appears on the All Filters story.

File: src/Filter/modules/DoubleMultiSelectFilter/DoubleMultiSelectFilter.tsx (hunk 2 above)
Comment (JonTheBurley) [reply src/Filter/modules/DoubleMultiSelectFilter/DoubleMultiSelectFilter.tsx:13]: Sure, I don't usually for only a few lines of styles but might as well be consistent and keep the file focused.

Comment (JonTheBurley) [reply src/Filter/modules/DoubleMultiSelectFilter/DoubleMultiSelectFilter.tsx]: Done.

**Comment by JonTheBurley** [comment]
> Looks good! Please add to the filterPageUtil.tsx so that it appears on the All Filters story.

I knew I was forgetting to add this filter somewhere. Thanks! Done.

**Comment by jason-ransom-slalom** [comment]
> > Looks good! Please add to the filterPageUtil.tsx so that it appears on the All Filters story.
> 
> I knew I was forgetting to add this filter somewhere. Thanks! Done.

Thank you!

**Review by adorack-tri — APPROVED** [review]

**Review by amirkibbar — APPROVED** [review]

**Pull request merged**